src/components/Products/ProductDetail/ProductDetail.component.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import { useSingleProduct } from '../../../utils/hooks/useSingleProduct';
+import Loading from '../../Loading';
+import Error from '../../Error';
+import ProductGallery from '../ProductGallery';
+import {
+  ProductWrapper,
+  Detail,
+  ColDetail,
+  ProductContent,
+  Title,
+  Price,
+  PurchaseInfo,
+  ProductSpecs,
+} from './ProductDetail.styles';
+
+const ProductDetail = () => {
+  const { id } = useParams();
+
+  //fetching data
+  const { data: singleProductData, isLoading } = useSingleProduct(id);
+  const { results: product } = singleProductData;
+
+  if (isLoading) {
+    return <Loading />;
+  }
+
+  if (!product && !isLoading) {
+    return <Error />;
+  }
+
+  if (!product) {
+    return null;
```

Comment:
this code is unreachable, if product is null it would always enter the previous if statement.